src/gui_common/IconProgressBar.cs

```diff
@@ -0,0 +1,55 @@
+using Godot;
+
+public class IconProgressBar : ColorRect
+{
+    public bool Disabled { get; private set; }
+    public int Location { get; private set; }
+    public int ActualLocation { get; private set; }
+
+    public void SetBarName(string name)
+    {
+        Name = name;
+    }
+
+    public void SetBarColour(Color colour)
+    {
+        Color = colour;
+    }
+
+    public void SetBarLeftShift(float leftShift)
+    {
+        MarginLeft = leftShift;
+    }
```

Comment:
These don't need setters. These are public things of the base class, they don't need to be encapsulated like this.